file/validate.go

```diff
@@ -1,13 +1,66 @@
 package file
 
 import (
+	"encoding/json"
 	"fmt"
 
 	ghodss "github.com/ghodss/yaml"
 	"github.com/kong/deck/utils"
 	"github.com/xeipuuv/gojsonschema"
 )
 
+func check(item string, t *string) bool {
+	if t != nil || len(*t) > 0 {
+		fmt.Println(item + "[err] is not allowed to be specified.")
+		return false
+	}
+	return true
+}
+
+func checkDefaults(kd KongDefaults) bool {
+	if ret := check("Service.ID", kd.Service.ID); !ret {
```

Comment:
This and every invocation to check has a nil exception. kd.Service is a pointer, which can be nil. Please add appropriate nil checks